File: scolary_v2/app/schemas/constituent_element.py
# begin #
# ---write your code here--- #
# end #

from datetime import datetime, time, date
from typing import Any
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, field_validator
from .journey import Journey


class ConstituentElementBase(BaseModel):
    name: Optional[str] = None
    semester: Optional[str] = None
    id_journey: Optional[int] = None
    color: Optional[str] = None


class ConstituentElementCreate(ConstituentElementBase):
    name: str


class ConstituentElementUpdate(ConstituentElementBase):
    pass


class ConstituentElementInDBBase(ConstituentElementBase):
    id: Optional[int]
    id_journey: Optional[int]

    model_config = ConfigDict(from_attributes=True)


class ConstituentElement(ConstituentElementInDBBase):
    pass


class ConstituentElementWithRelation(ConstituentElementInDBBase):
    journey: Optional[Journey] = None


class ConstituentElementInDB(ConstituentElementInDBBase):
    pass


class ResponseConstituentElement(BaseModel):
    count: int
    data: Optional[List[ConstituentElementWithRelation]]


# begin #
# ---write your code here--- #
# end #

File: scolary-react/src/models/constituent-element.ts
import { Journey } from "./journey";

export interface ConstituentElement {
  id: number;
  name: string;
  semester: string;
  color: string;
  id_journey: number;
  journey?: Journey | null;
}

export type ConstituentElementPayload = Pick<
  ConstituentElement,
  "name" | "semester" | "id_journey" | "color"
> & { id_journey: number | string };

export type ConstituentElementListQuery = Record<
  string,
  string | number | boolean | undefined
>;
